include/circt/Tools/arcilator/pipelines.h

```diff
@@ -0,0 +1,98 @@
+#ifndef TOOLS_ARCILATOR_PIPELINES_H
+#define TOOLS_ARCILATOR_PIPELINES_H
+
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Pass/PassOptions.h"
+#include "llvm/Support/CommandLine.h"
+
+namespace circt {
+
+// Pre-process the input such that it no longer contains any SV dialect ops
+// and external modules that are relevant to the arc transformation are
+// represented as intrinsic ops.
+struct ArcPreprocessingOptions
+    : mlir::PassPipelineOptions<ArcPreprocessingOptions> {
+  Option<bool> observePorts{*this, "observe-ports",
+                            llvm::cl::desc("Make all ports observable"),
+                            llvm::cl::init(false)};
+
+  Option<bool> observeWires{*this, "observe-wires",
+                            llvm::cl::desc("Make all wires observable"),
+                            llvm::cl::init(false)};
+
+  Option<bool> observeNamedValues{
+      *this, "observe-named-values",
+      llvm::cl::desc("Make values with `sv.namehint` observable"),
+      llvm::cl::init(false)};
+
+  Option<bool> observeMemories{
+      *this, "observe-memories",
+      llvm::cl::desc("Make all memory contents observable"),
+      llvm::cl::init(false)};
+};
+void arcPreprocessingPipeline(mlir::PassManager &pm,
```

Comment:
I have slight preference to keeping the populate prefix/verb for these TBH. This appends passes to existing passmanager rather than returning a new pipeline just around preproccesing for example. It is used consistently for patterns, but pipelines seem to vary between build, add, and populate.